File: Time Speed Test Python/time speed test.py
import datetime
i = 0
a = 2
b = 1
c = 0
start1 = datetime.datetime.now()

while i <= 1000000:
    i = i + 1
    c = (a - b)*(a + b)

end1 = datetime.datetime.now()
diffrent1 = start1 - end1
c = 0
start2 = datetime.datetime.now()

while i != 0:
    i = i - 1
    c = a**2 - b**2

end2 = datetime.datetime.now()
diffrent2 = start2 - end2
c = 0 
start3 = datetime.datetime.now()

while i <= 1000000:
    i = i + 1
    c = (a << 1) -( b << 1)

end3 = datetime.datetime.now()
diffrent3 = start3 - end3

print ("start 1 = ", start1)
print ("end 1 = ", end1)
print ("start 2 = ", start2)
print ("end 2 = ", end2)
print ("difference 1 = ",diffrent1.microseconds)
print ("difference 2 = ",diffrent2.microseconds)
print ("difference 3 = ",diffrent3.microseconds)
print ("difference 2 - difference 1 = ", diffrent2.microseconds - diffrent1.microseconds)
print ("difference 3 - difference 2 = ", diffrent3.microseconds - diffrent2.microseconds)

File: Time Speed Test Cpp/Time Speed Test/Time Speed Test.cpp
#include <iostream>
#include <cmath>
#include <ctime>
using namespace std;

int main()
{
	time_t start1, start2, start3;
	time_t end1, end2, end3;
	time_t difference1, difference2, difference3;
	int a = 2;
	int b = 1;
	int c = 0;
	start1 = time(NULL) * 1000000;
	for (int i = 0; i <= 1000000; i++)
	{
		c = (a - b) * (a + b);
	}
	end1 = time(NULL) * 1000000;
	difference1 = end1 - start1;
	start2 = time(NULL) * 1000000;
	for (int i = 0; i <= 1000000; i++)
	{
		c = pow(a, 2) + pow(b, 2);
	}
	end2 = time(NULL) * 1000000;
	difference2 = end2 - start2;
	start3 = time(NULL) * 1000000;
	for (int i = 0; i <= 1000000; i++)
	{
		c = (a << 1) - (b << 1);
	}
	end3 = time(NULL) * 1000000;
	difference3 = end3 - start3;
	cout << "start 1 (microseconds) = " << start1 << endl;
	cout << "start 2 (microseconds) = " << start2 << endl;
	cout << "start 3 (microseconds) = " << start3 << endl;
	cout << "end 1 (microseconds) = " << end1 << endl;
	cout << "end 2 (microseconds) = " << end2 << endl;
	cout << "end 3 (microseconds) = " << end3 << endl;
	cout << endl;
	cout << "difference = end - start" << endl;
	cout << endl;
	cout << "difference 1 = " << difference1 << endl;
	cout << "difference 2 = " << difference2 << endl;
	cout << "difference 3 = " << difference3 << endl;
	system("pause");
	return 0;
}